frontend/tests/setup/window.ts

```diff
@@ -0,0 +1,18 @@
+/*
+    WalletConnect's modal library uses the window object for detecting mobile devices
+    therefore a window mock is required globally
+*/
+
+Object.defineProperty(window, 'matchMedia', {
```

Comment:
The problem here was that the tests for all of the files that import from `@/services/web3-provider/util-export.ts` are failing due to the lack of mock on `window.matchMedia`.
This also includes the files that depend on the providers: `useWallet.ts`, `WalletMenu.vue` etc. 
So i thought of adding this mock globally instead of including it in all test files where needed.

 Two possible ways to address this:
- Using dynamic imports inside `@/services/web3-provider/util-export.ts`? 
The drawback of this approach is that it increases complexity in the mocking process.
- Move the imports from `@/services/web3-provider/util-export.ts` into their respective provider files and try to find a different way to mock them inside the tests. The reason why i created `util-export.ts` a while ago is because I faced some issues when mocking the libraries. To be more specific, some of the libraries had only default exports and no named exports which made the mocking process difficult.

If you think this is a hard requirement i could start with the 2nd option and see how it goes.

Reply:
Don't take too much effort on this. Since it is more files that need this, just leave it like it is. I thought there would an easy option to just provide the mock in the test files in which it is required, but it sounds much more complex if it is needed on import.